From d903b4b98ef6f28e5cf9890c9619f3de11b80815 Mon Sep 17 00:00:00 2001
From: Chaitanya Vadrevu <chaitanya.vadrevu@ni.com>
Date: Tue, 14 Jan 2025 14:41:06 -0600
Subject: [PATCH] setup.py: Changes to get build working without rust and with
 python 3.5

---
 setup.py | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 74f69e7..30a1dea 100644
--- a/setup.py
+++ b/setup.py
@@ -11,7 +11,8 @@ import sys
 from setuptools import find_packages, setup
 
 try:
-    from setuptools_rust import RustExtension
+    if not os.environ.get("CRYPTOGRAPHY_DONT_BUILD_RUST"):
+        from setuptools_rust import RustExtension
 except ImportError:
     print(
         """
@@ -41,8 +42,8 @@ with open(os.path.join(src_dir, "cryptography", "__about__.py")) as f:
 
 # `install_requirements` and `setup_requirements` must be kept in sync with
 # `pyproject.toml`
-setuptools_rust = "setuptools-rust>=0.11.4"
-install_requirements = ["cffi>=1.12"]
+setuptools_rust = ""
+install_requirements = ["cffi>=1.8"]
 setup_requirements = install_requirements + [setuptools_rust]
 
 if os.environ.get("CRYPTOGRAPHY_DONT_BUILD_RUST"):
@@ -92,6 +93,7 @@ try:
             "Programming Language :: Python",
             "Programming Language :: Python :: 3",
             "Programming Language :: Python :: 3 :: Only",
+            "Programming Language :: Python :: 3.5",
             "Programming Language :: Python :: 3.6",
             "Programming Language :: Python :: 3.7",
             "Programming Language :: Python :: 3.8",
@@ -105,7 +107,7 @@ try:
             where="src", exclude=["_cffi_src", "_cffi_src.*"]
         ),
         include_package_data=True,
-        python_requires=">=3.6",
+        python_requires=">=3.5",
         install_requires=install_requirements,
         setup_requires=setup_requirements,
         extras_require={
